### PR TITLE
OpenShift cost report

### PR DESCRIPTION
**Pull request by hemslo**
Add `openshift` cost report, similar to `aws` cost report.

Main differences:

* api endpoint for [cost management api](https://github.com/app-sre/qontract-reconcile/pull/4305/files#diff-e61cb1355015db6a6ec36175babebcaee1ba0e4287b2eeff74e6fb8f9368449c).
* total and delta calculation is aggregated in `app` level in code instead of server side, because cost management has no direct link between `app` and `namespace` costs. [code](https://github.com/app-sre/qontract-reconcile/pull/4305/files#diff-4f28501a8ecb3be2b8bf4c085c15d7f19f2841a0633c749e4bedd614da2ba854R99-R155)

Command to generate report:

```shell
qontract-cli --config config.toml get openshift-cost-report
```

Depends on schema change https://github.com/app-sre/qontract-schemas/pull/643

`cost optimization` report will be done in a new PR.

[APPSRE-10104](https://issues.redhat.com/browse/APPSRE-10104)

### CODE REVIEW

File: tools/cli_commands/cost_report/openshift.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    def get_apps(self) -> list[App]:
+        return get_app_names(self.gql_api)
+
+    def get_cost_namespaces(self) -> list[CostNamespace]:
+        return get_cost_namespaces(self.gql_api)
+
+    def _get_report(
+        self,
+        cost_namespace: CostNamespace,
+    ) -> Tuple[CostNamespace, OpenShiftReportCostResponse]:
```
Comment (chassing) [inline tools/cli_commands/cost_report/openshift.py:53]:
```suggestion
    ) -> tuple[CostNamespace, OpenShiftReportCostResponse]:
```

File: tools/cli_commands/cost_report/openshift.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+from reconcile.typed_queries.cost_report.settings import get_cost_report_settings
+from reconcile.utils import gql
+from reconcile.utils.secret_reader import create_secret_reader
+from tools.cli_commands.cost_report.cost_management_api import CostManagementApi
+from tools.cli_commands.cost_report.model import ChildAppReport, Report, ReportItem
+from tools.cli_commands.cost_report.response import OpenShiftReportCostResponse
+from tools.cli_commands.cost_report.util import process_reports
+from tools.cli_commands.cost_report.view import render_openshift_cost_report
+
+THREAD_POOL_SIZE = 10
```
Comment (chassing) [inline tools/cli_commands/cost_report/openshift.py:25]: nit: The thread pool size should be configurable via `__init__`

**Review by chassing — APPROVED** [review]

File: tools/cli_commands/cost_report/openshift.py (hunk 1 above)
Comment (hemslo) [reply tools/cli_commands/cost_report/openshift.py]: replaced all deprecated alias at https://github.com/app-sre/qontract-reconcile/pull/4305/commits/b79818da8f5085167ede1edfb8802d2f2b0dbe44

File: tools/cli_commands/cost_report/openshift.py (hunk 2 above)
Comment (hemslo) [reply tools/cli_commands/cost_report/openshift.py:25]: the way we call this command has no input options, so I didn't make it configurable at cli level, only added init params at https://github.com/app-sre/qontract-reconcile/pull/4305/commits/24d6a672439016d0ee8ea5ccf30bfa58079d6336

**Pull request merged**